Automattic/wp-calypso
# Pattern Assembler: Improve a11y and keyboard navigation

Related to https://github.com/Automattic/wp-calypso/issues/74188, https://github.com/Automattic/wp-calypso/issues/74189

## Proposed Changes

* Focus on the first tabbable element when you navigate to another screen
* Enable the focus restoration when you navigate back to the previous screen. It looks good after we upgraded `@wordpress/components`
* Support keyboard navigation (arrow keys) on the options of the list (e.g. Categories, Patterns, Variations)

https://user-images.githubusercontent.com/13596067/233023019-d6cc98d0-72b1-412b-9bc8-d37395c68857.mov

## Testing Instructions



* Go to /setup/site-setup?siteSlug=<your_free_site>
* Continue until you land on the Design Picker
* On the Design Picker screen, scroll down to the bottom and select BCPA CTA
* On the Pattern Assembler screen
  * Use the keyboard to select the header, sections, footer, colors, and fonts
  * Ensure it's friendly to assemble your homepage via the keyboard on the sidebar

## Pre-merge Checklist
